site/tests/phpstan/SubmittyMagicVariant.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace tests\phpstan;
+
+use PHPStan\Reflection\ParametersAcceptor;
+use PHPStan\Type\Generic\TemplateTypeMap;
+use PHPStan\Type\Type;
+
+class SubmittyMagicVariant implements ParametersAcceptor {
+    private array $params;
+
+    private Type $returnType;
+
+    public function __construct(array $params, Type $returnType) {
+        $this->params = $params;
+        $this->returnType = $returnType;
+    }
+
+
+    public function getTemplateTypeMap(): TemplateTypeMap {
+        return TemplateTypeMap::createEmpty();
```

Comment:
Is this necessarily empty?

Reply:
Tbh I couldn't really find what this gets used for. If you have any ideas, please lmk.

Reply:
It's surprisingly difficult to find documentation about this.  The best example I can find is https://github.com/phpstan/phpstan/issues/6371.  If I'm understanding this correctly, the use case we'd want to support is automatic templating support for an attribute defined like this:
```
/** @var array<T> $foo */
public array $foo;
```